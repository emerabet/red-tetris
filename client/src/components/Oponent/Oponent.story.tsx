import React from 'react';
import { storiesOf } from '@storybook/react';
import Oponent from './Oponent';

storiesOf('Oponent', module)
  .add('one oponent', () => (
    <Oponent
    spectre={
      {
        id: 'oponent 1',
        spectre: '0123456789',
        username: 'aaaaa',
      }
    }
    />
  ));
