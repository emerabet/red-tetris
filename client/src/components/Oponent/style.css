.opponentFlex {
    width: 50%;
    display: flex;
    /* flex-direction: column; */
    /* align-items: center; */
    /* justify-content: center; */
}

.opponent {
    width: 74px;
    height: 144px;
    border: solid black 2px;
    margin-top: 10px;
    display: flex;
    flex-direction: row;
    /* align-items: flex-end; */
}

.opponentColumn {
    background: #cfd8dc;
    -webkit-box-shadow: inset 0 0 1px 0 #000;
  box-shadow: inset 0 0 1px 0 #000;
}

.opponentColumn_0 {
    width: 7px;
    height: 0px;
}

.opponentColumn_1 {
    width: 7px;
    height: 7px;
}

.opponentColumn_2 {
    width: 7px;
    height: 14px;
}

.opponentColumn_3 {
    width: 7px;
    height: 21px;
}

.opponentColumn_4 {
    width: 7px;
    height: 28px;
}

.opponentColumn_5 {
    width: 7px;
    height: 35px;
}

.opponentColumn_6 {
    width: 7px;
    height: 42px;
}

.opponentColumn_7 {
    width: 7px;
    height: 49px;
}

.opponentColumn_8 {
    width: 7px;
    height: 56px;
}

.opponentColumn_9 {
    width: 7px;
    height: 63px;
}

.opponentColumn_A {
    width: 7px;
    height: 70px;
}

.opponentColumn_B {
    width: 7px;
    height: 77px;
}

.opponentColumn_C {
    width: 7px;
    height: 84px;
}

.opponentColumn_D {
    width: 7px;
    height: 91px;
}

.opponentColumn_E {
    width: 7px;
    height: 98px;
}

.opponentColumn_F {
    width: 7px;
    height: 105px;
}

.opponentColumn_G {
    width: 7px;
    height: 112px;
}

.opponentColumn_H {
    width: 7px;
    height: 119px;
}

.opponentColumn_I {
    width: 7px;
    height: 126px;
}

.opponentColumn_J {
    width: 7px;
    height: 133px;
}

.opponentColumn_K {
    width: 7px;
    height: 140px;
}

.opponentFull {
    background: black;
  -webkit-box-shadow: inset 0 0 1px 1px #cfd8dc;
  box-shadow: inset 0 0 1px 1px #cfd8dc;
}