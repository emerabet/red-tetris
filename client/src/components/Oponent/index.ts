import Oponent from './Oponent';

export default Oponent;
