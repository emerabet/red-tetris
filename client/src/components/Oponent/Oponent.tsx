import React from 'react';
import SquareLabel from '../Labels';

import { SpectreI } from '../../types/gameTypes';

import './style.css';

interface OponentProps {
  spectre: SpectreI;
}

const Oponent: React.SFC<OponentProps> = (props) => {
  const game: string[] = props.spectre.spectre.split('');
  return (
    <div className="opponentFlex">
      <div className="opponent">
        {
          game.map((g, i) => {
            return (
              <div key={`op_${i}`} className={`opponentColumn opponentColumn_${g}`}></div>
            );
          })
        }
      </div>
      <SquareLabel
        label={props.spectre.id}
        white
        additionalClassName="small"
      />
    </div>

  );
};

export default Oponent;
