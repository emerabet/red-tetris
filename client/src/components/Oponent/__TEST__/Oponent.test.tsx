import React from 'react';

import renderer from 'react-test-renderer';
import Oponent from '../Oponent';

it('renders correctly with defaults', () => {
  const oponent = renderer.create(<Oponent
    spectre = {{
        id: 'oponent 1',
        spectre: '0123456789',
        username: 'aaaaa',
      }
    }
     />).toJSON();
  expect(oponent).toMatchSnapshot();
});
