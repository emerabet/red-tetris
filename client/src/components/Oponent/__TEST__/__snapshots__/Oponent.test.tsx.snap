// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="opponentFlex"
>
  <div
    className="opponent"
  >
    <div
      className="opponentColumn opponentColumn_0"
    />
    <div
      className="opponentColumn opponentColumn_1"
    />
    <div
      className="opponentColumn opponentColumn_2"
    />
    <div
      className="opponentColumn opponentColumn_3"
    />
    <div
      className="opponentColumn opponentColumn_4"
    />
    <div
      className="opponentColumn opponentColumn_5"
    />
    <div
      className="opponentColumn opponentColumn_6"
    />
    <div
      className="opponentColumn opponentColumn_7"
    />
    <div
      className="opponentColumn opponentColumn_8"
    />
    <div
      className="opponentColumn opponentColumn_9"
    />
  </div>
  <div
    className="labelTitle small"
  >
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube o1"
        />
        <div
          className="labelCube whiteCube o2"
        />
        <div
          className="labelCube whiteCube o3"
        />
        <div
          className="labelCube whiteCube o4"
        />
        <div
          className="labelCube whiteCube o5"
        />
        <div
          className="labelCube whiteCube o6"
        />
        <div
          className="labelCube whiteCube o7"
        />
        <div
          className="labelCube whiteCube o8"
        />
        <div
          className="labelCube whiteCube o9"
        />
        <div
          className="labelCube whiteCube o10"
        />
        <div
          className="labelCube whiteCube o11"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube p1"
        />
        <div
          className="labelCube whiteCube p2"
        />
        <div
          className="labelCube whiteCube p3"
        />
        <div
          className="labelCube whiteCube p4"
        />
        <div
          className="labelCube whiteCube p5"
        />
        <div
          className="labelCube whiteCube p6"
        />
        <div
          className="labelCube whiteCube p7"
        />
        <div
          className="labelCube whiteCube p8"
        />
        <div
          className="labelCube whiteCube p9"
        />
        <div
          className="labelCube whiteCube p10"
        />
        <div
          className="labelCube whiteCube p11"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube o1"
        />
        <div
          className="labelCube whiteCube o2"
        />
        <div
          className="labelCube whiteCube o3"
        />
        <div
          className="labelCube whiteCube o4"
        />
        <div
          className="labelCube whiteCube o5"
        />
        <div
          className="labelCube whiteCube o6"
        />
        <div
          className="labelCube whiteCube o7"
        />
        <div
          className="labelCube whiteCube o8"
        />
        <div
          className="labelCube whiteCube o9"
        />
        <div
          className="labelCube whiteCube o10"
        />
        <div
          className="labelCube whiteCube o11"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube n1"
        />
        <div
          className="labelCube whiteCube n2"
        />
        <div
          className="labelCube whiteCube n3"
        />
        <div
          className="labelCube whiteCube n4"
        />
        <div
          className="labelCube whiteCube n5"
        />
        <div
          className="labelCube whiteCube n6"
        />
        <div
          className="labelCube whiteCube n7"
        />
        <div
          className="labelCube whiteCube n8"
        />
        <div
          className="labelCube whiteCube n9"
        />
        <div
          className="labelCube whiteCube n10"
        />
        <div
          className="labelCube whiteCube n11"
        />
        <div
          className="labelCube whiteCube n12"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube e1"
        />
        <div
          className="labelCube whiteCube e2"
        />
        <div
          className="labelCube whiteCube e3"
        />
        <div
          className="labelCube whiteCube e4"
        />
        <div
          className="labelCube whiteCube e5"
        />
        <div
          className="labelCube whiteCube e6"
        />
        <div
          className="labelCube whiteCube e7"
        />
        <div
          className="labelCube whiteCube e8"
        />
        <div
          className="labelCube whiteCube e9"
        />
        <div
          className="labelCube whiteCube e10"
        />
        <div
          className="labelCube whiteCube e11"
        />
        <div
          className="labelCube whiteCube e12"
        />
        <div
          className="labelCube whiteCube e13"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube n1"
        />
        <div
          className="labelCube whiteCube n2"
        />
        <div
          className="labelCube whiteCube n3"
        />
        <div
          className="labelCube whiteCube n4"
        />
        <div
          className="labelCube whiteCube n5"
        />
        <div
          className="labelCube whiteCube n6"
        />
        <div
          className="labelCube whiteCube n7"
        />
        <div
          className="labelCube whiteCube n8"
        />
        <div
          className="labelCube whiteCube n9"
        />
        <div
          className="labelCube whiteCube n10"
        />
        <div
          className="labelCube whiteCube n11"
        />
        <div
          className="labelCube whiteCube n12"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube e1"
        />
        <div
          className="labelCube whiteCube e2"
        />
        <div
          className="labelCube whiteCube e3"
        />
        <div
          className="labelCube whiteCube e4"
        />
        <div
          className="labelCube whiteCube e5"
        />
        <div
          className="labelCube whiteCube e6"
        />
        <div
          className="labelCube whiteCube e7"
        />
        <div
          className="labelCube whiteCube e8"
        />
        <div
          className="labelCube whiteCube e9"
        />
        <div
          className="labelCube whiteCube e10"
        />
        <div
          className="labelCube whiteCube e11"
        />
        <div
          className="labelCube whiteCube e12"
        />
        <div
          className="labelCube whiteCube e13"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetter"
      >
        <div
          className="labelCube whiteCube t1"
        />
        <div
          className="labelCube whiteCube t2"
        />
        <div
          className="labelCube whiteCube t3"
        />
        <div
          className="labelCube whiteCube t4"
        />
        <div
          className="labelCube whiteCube t5"
        />
        <div
          className="labelCube whiteCube t6"
        />
        <div
          className="labelCube whiteCube t7"
        />
        <div
          className="labelCube whiteCube t8"
        />
        <div
          className="labelCube whiteCube t9"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube e1"
        />
        <div
          className="labelCube whiteCube e2"
        />
        <div
          className="labelCube whiteCube e3"
        />
        <div
          className="labelCube whiteCube e4"
        />
        <div
          className="labelCube whiteCube e5"
        />
        <div
          className="labelCube whiteCube e6"
        />
        <div
          className="labelCube whiteCube e7"
        />
        <div
          className="labelCube whiteCube e8"
        />
        <div
          className="labelCube whiteCube e9"
        />
        <div
          className="labelCube whiteCube e10"
        />
        <div
          className="labelCube whiteCube e11"
        />
        <div
          className="labelCube whiteCube e12"
        />
        <div
          className="labelCube whiteCube e13"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetter"
      >
        <div
          className="labelCube whiteCube t1"
        />
        <div
          className="labelCube whiteCube t2"
        />
        <div
          className="labelCube whiteCube t3"
        />
        <div
          className="labelCube whiteCube t4"
        />
        <div
          className="labelCube whiteCube t5"
        />
        <div
          className="labelCube whiteCube t6"
        />
        <div
          className="labelCube whiteCube t7"
        />
        <div
          className="labelCube whiteCube t8"
        />
        <div
          className="labelCube whiteCube t9"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube e1"
        />
        <div
          className="labelCube whiteCube e2"
        />
        <div
          className="labelCube whiteCube e3"
        />
        <div
          className="labelCube whiteCube e4"
        />
        <div
          className="labelCube whiteCube e5"
        />
        <div
          className="labelCube whiteCube e6"
        />
        <div
          className="labelCube whiteCube e7"
        />
        <div
          className="labelCube whiteCube e8"
        />
        <div
          className="labelCube whiteCube e9"
        />
        <div
          className="labelCube whiteCube e10"
        />
        <div
          className="labelCube whiteCube e11"
        />
        <div
          className="labelCube whiteCube e12"
        />
        <div
          className="labelCube whiteCube e13"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetter"
      >
        <div
          className="labelCube whiteCube t1"
        />
        <div
          className="labelCube whiteCube t2"
        />
        <div
          className="labelCube whiteCube t3"
        />
        <div
          className="labelCube whiteCube t4"
        />
        <div
          className="labelCube whiteCube t5"
        />
        <div
          className="labelCube whiteCube t6"
        />
        <div
          className="labelCube whiteCube t7"
        />
        <div
          className="labelCube whiteCube t8"
        />
        <div
          className="labelCube whiteCube t9"
        />
      </div>
       
    </div>
    <div>
       
      <div
        className="labelLetterM"
      >
        <div
          className="labelCube whiteCube e1"
        />
        <div
          className="labelCube whiteCube e2"
        />
        <div
          className="labelCube whiteCube e3"
        />
        <div
          className="labelCube whiteCube e4"
        />
        <div
          className="labelCube whiteCube e5"
        />
        <div
          className="labelCube whiteCube e6"
        />
        <div
          className="labelCube whiteCube e7"
        />
        <div
          className="labelCube whiteCube e8"
        />
        <div
          className="labelCube whiteCube e9"
        />
        <div
          className="labelCube whiteCube e10"
        />
        <div
          className="labelCube whiteCube e11"
        />
        <div
          className="labelCube whiteCube e12"
        />
        <div
          className="labelCube whiteCube e13"
        />
      </div>
       
    </div>
  </div>
</div>
`;
